### PR TITLE
Bump kubernetes to v1.16.15

### PR DESCRIPTION
**Pull request by KanShiori**
### What problem does this PR solve?

bump kubernetes to v1.16.15

### What is changed and how does it work?

* exec cmd `hack/pin-deps,sh v1.16.15`
* bump kubernetes to v1.16.15 in submodules

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [ ] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [x] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Bump kubernetes to v1.16.15
```

### CODE REVIEW

File: hack/pin-deps.sh [hunk 1]
```diff
@@ -57,8 +57,8 @@ edit_args=(
 )
 
 for repo in ${STAGING_REPOS[@]}; do
-    if version_ge "1.17.0" $VERSION; then
-        echo ">=1.17.0"
+    if version_ge "1.16.3" $VERSION; then
```
Comment (DanielZhangQD) [inline hack/pin-deps.sh:60]: Why change this?

**Review by july2993 — APPROVED** [review]

File: hack/pin-deps.sh (hunk 1 above)
Comment (KanShiori) [reply hack/pin-deps.sh:60]: The old script import libs with version `$repo@kubernetes-$VERSION`  before v1.17.0. But libs already have semantic versions which can be imported after v1.16.3.

File: Makefile [hunk 2]
```diff
@@ -178,10 +177,9 @@ lint:
 	./hack/verify-lint.sh
 
 tidy:
-	@echo "go mod tidy"
-	for dir in . $(GO_SUBMODULE_DIRS); do \
-		cd $$dir && go mod tidy && git diff -U --exit-code go.mod go.sum && cd -; \
-	done
+	go mod tidy && git diff -U --exit-code go.mod go.sum
+	cd pkg/apis && go mod tidy && git diff -U --exit-code go.mod go.sum
+	cd pkg/client && go mod tidy && git diff -U --exit-code go.mod go.sum
```
Comment (KanShiori) [inline Makefile:182]: There seems something wrong when execute cmd `cd` in a loop in Makefile :cold_sweat:, so I change it

Comment (DanielZhangQD) [reply Makefile:182]: No need to change to the current dir after L181?

Comment (KanShiori) [reply Makefile:182]: It works in my VM, but fail in CI

**Pull request closed without merging**